languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py

```diff
@@ -16,41 +17,55 @@
 EmitFunction = Callable[[Session, Any], Any]
 
 
-def partial_to_filter(expression: Expression, session: Session, model):
+def partial_to_filter(expression: Expression, session: Session, model, get_model):
     """Convert constraints in ``partial`` to a filter over ``model`` that should be applied to query."""
-    return translate_expr(expression, session, model)
+    return translate_expr(expression, session, model, get_model)
 
 
 # Returns None or the translated expression.
-def translate_expr(expression: Expression, session: Session, model):
+def translate_expr(expression: Expression, session: Session, model, get_model):
     assert isinstance(expression, Expression)
     if expression.operator == "Eq" or expression.operator == "Unify":
-        return translate_compare(expression, session, model)
+        return translate_compare(expression, session, model, get_model)
     elif expression.operator == "Isa":
-        assert expression.args[1].tag == model.__name__
-        return None
+        return translate_isa(expression, session, model, get_model)
     elif expression.operator == "In":
-        return translate_in(expression, session, model)
+        return translate_in(expression, session, model, get_model)
     elif expression.operator == "And":
-        return translate_and_expr(expression, session, model)
+        return translate_and(expression, session, model, get_model)
     else:
         raise UnsupportedError(f"Unsupported {expression}")
 
 
-def translate_and_expr(expression: Expression, session: Session, model):
+def translate_and(expression: Expression, session: Session, model, get_model):
     assert expression.operator == "And"
     expr = and_()
     for expression in expression.args:
-        translated = translate_expr(expression, session, model)
+        translated = translate_expr(expression, session, model, get_model)
         if translated is None:
             continue
 
         expr = expr & translated
 
     return expr
 
+def translate_isa(expression: Expression, session: Session, model, get_model):
+    assert expression.operator == "Isa"
+    left, right = expression.args
+    if dot_path(left) == ():
+        assert left == Variable("_this")
+    else:
+        for field_name in dot_path(left):
+            _, model, __ = get_relationship(model, field_name)
+
+    assert not right.fields, "Unexpected fields in isa expression"
+    constraint_type = get_model(right.tag)
+    if not issubclass(model, constraint_type):
+        return sql.false()
+    else:
+        return None
```

Comment:
Also is there a `sql.true()` (or equivalent) that we could call here instead of `None`? @plotnick and I just made the same change in the Odoo adapter: https://github.com/osohq/oso_odoo/pull/4/files#diff-6d3cd0165f6f824122a73df7d8e31e00e5479c533ed967833761eb1888858023L66-R70
```suggestion
    if issubclass(model, constraint_type):
        return None
    else:
        return sql.false()
```